internal/generator/vector/output/kafka/kafka.go

```diff
@@ -52,34 +48,45 @@ func (k *Kafka) SetCompression(algo string) {
 	k.Compression.Value = algo
 }
 
-func New(id string, o logging.OutputSpec, inputs []string, secret *corev1.Secret, strategy common.ConfigStrategy, op Options) []Element {
+func New(id string, o obs.OutputSpec, inputs []string, secrets vectorhelpers.Secrets, strategy common.ConfigStrategy, op Options) []Element {
 	if genhelper.IsDebugOutput(op) {
 		return []Element{
 			Debug(id, vectorhelpers.MakeInputs(inputs...)),
 		}
 	}
 
-	brokers, genTlsConf := Brokers(o)
-	sink := Output(id, o, inputs, secret, op, brokers)
+	brokers := Brokers(o)
+	sink := sink(id, o, inputs, op, brokers)
 	if strategy != nil {
 		strategy.VisitSink(sink)
 	}
-	return MergeElements(
-		[]Element{
-			sink,
-			Encoding(id, op),
-			common.NewAcknowledgments(id, strategy),
-			common.NewBatch(id, strategy),
-			common.NewBuffer(id, strategy),
-		},
-		TLSConf(id, o, secret, op, genTlsConf),
-		SASLConf(id, o, secret),
+	tlsConfig := []Element{Nil}
+	if o.TLS != nil {
+		skipVerify := o.TLS.InsecureSkipVerify
+		o.TLS.InsecureSkipVerify = false
+		tlsConfig = []Element{tls.New(id, o.TLS, secrets, op)}
+		if skipVerify {
+			tlsConfig = append(tlsConfig, InsecureTLS{
+				ComponentID: id,
+			})
+		}
+	}
+	elements := []Element{
+		sink,
+		Encoding(id, op),
+		common.NewAcknowledgments(id, strategy),
+		common.NewBatch(id, strategy),
+		common.NewBuffer(id, strategy),
+		SASLConf(id, o.Kafka.Authentication, secrets),
+	}
+	elements = append(elements,
```

Comment:
Should we keep using the `MergeElements()` function?

Reply:
:man_shrugging:   I'm indifferent to its use.  It's not any cleaner IMO but pleas use it when you feel it's necessary